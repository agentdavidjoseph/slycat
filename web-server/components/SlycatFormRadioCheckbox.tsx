'use strict';
import * as React from 'react';

/**
 */
export interface SlycatFormRadioCheckboxProps {
  checked: boolean
  onChange: Function
  value: string
  text: string
  style: any
}

/**
 * not used
 */
export interface SlycatFormRadioCheckboxState {
}
/**
 * class that creates a a form with checkboxes
 * some other process
 */
export default class SlycatFormRadioCheckbox extends React.Component<SlycatFormRadioCheckboxProps, SlycatFormRadioCheckboxState> {
  /**
   * not used
   */
  public constructor(props:SlycatFormRadioCheckboxProps) {
    super(props)
    this.state = {}
  }

  public render () {
    return (
      <div className="form-check">
        <label className="form-check-label" style={this.props.style} htmlFor="radio1">
          <input type="radio" className="form-check-input" value={this.props.value} checked={this.props.checked} onChange={(e)=>this.props.onChange(e.target.value)}/>
          {this.props.text}
        </label>
      </div>
    );
  }
}
