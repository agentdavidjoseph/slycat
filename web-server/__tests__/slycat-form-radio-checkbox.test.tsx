import React from 'react';
import SlycatFormRadioCheckbox, {SlycatFormRadioCheckboxProps, SlycatFormRadioCheckboxState} from "components/SlycatFormRadioCheckbox";
import { mount } from "enzyme";

describe('when loading a slycat selector',() =>{

  const properties: SlycatFormRadioCheckboxProps = {
    checked:true,
    onChange:jest.fn(),
    value:'local',
    text:'Local',
    style:{marginRight: '92%'}
  };
  const render = mount(
    <SlycatFormRadioCheckbox
      {...properties}
    />
  );
  test('we have expected props on initial load', () => {
    const props = render.props() as SlycatFormRadioCheckboxProps;
    expect(props).toMatchSnapshot();
  });

  test('we see no state', () => {
    const state = render.state() as SlycatFormRadioCheckboxState;
    expect(state).toMatchSnapshot();
  });

  test('we should have a full initialization', () => {
    expect(render.instance()).toMatchSnapshot();
  });

  test('we should onChange function get value', () => {
    const myEventHandler = jest.fn();
    render.setProps({ onChange: myEventHandler});
    render.find('input').simulate('change','local');
    expect(myEventHandler).toHaveBeenCalledWith('local');
  });
});

export default undefined