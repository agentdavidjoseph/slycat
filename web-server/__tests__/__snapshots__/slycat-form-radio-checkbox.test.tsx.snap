// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when loading a slycat selector we have expected props on initial load 1`] = `
Object {
  "checked": true,
  "onChange": [MockFunction],
  "style": Object {
    "marginRight": "92%",
  },
  "text": "Local",
  "value": "local",
}
`;

exports[`when loading a slycat selector we see no state 1`] = `Object {}`;

exports[`when loading a slycat selector we should have a full initialization 1`] = `
<SlycatFormRadioCheckbox
  checked={true}
  onChange={[MockFunction]}
  style={
    Object {
      "marginRight": "92%",
    }
  }
  text="Local"
  value="local"
>
  <div
    className="form-check"
  >
    <label
      className="form-check-label"
      htmlFor="radio1"
      style={
        Object {
          "marginRight": "92%",
        }
      }
    >
      <input
        checked={true}
        className="form-check-input"
        onChange={[Function]}
        type="radio"
        value="local"
      />
      Local
    </label>
  </div>
</SlycatFormRadioCheckbox>
`;
